libs/angular/request/src/lib/models/requested.type.ts

```diff
@@ -0,0 +1,11 @@
+export class RequestState {
+    static Empty = class Empty {};
+    static Loading = class Loading {};
+    static Error = class Error {};
+}
+
+export type Requested<T> =
+    | T
+    | typeof RequestState.Empty
+    | typeof RequestState.Loading
+    | typeof RequestState.Error;
```

Comment:
A `Requested` datum is defined by being either the loaded data or one of the possible `RequestState`s.